### PR TITLE
[Cpp]Get next job id globally instead of random selecting

### PR DESCRIPTION
**Pull request by WangTaoTheTonic**
## Related issue number
Final part of #13984

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by wuisawesome — APPROVED** [review]

File: cpp/src/ray/util/process_helper.cc [hunk 1]
```diff
@@ -102,11 +102,7 @@ void ProcessHelper::RayStart(CoreWorkerOptions::TaskExecutionCallback callback)
   if (log_dir.empty()) {
     std::string session_dir = ConfigInternal::Instance().session_dir;
     if (session_dir.empty()) {
-      if (!global_state_accessor) {
-        global_state_accessor = ProcessHelper::GetInstance().CreateGlobalStateAccessor(
-            redis_address, ConfigInternal::Instance().redis_password);
-        RAY_CHECK(global_state_accessor->Connect()) << "Failed to connect to GCS.";
-      }
+      RAY_CHECK(global_state_accessor->Connect()) << "Failed to connect to GCS.";
```
Comment (kfstorm) [inline cpp/src/ray/util/process_helper.cc:105]: Should we always call `global_state_accessor->Connect()` right after the accessor is created?

File: cpp/src/ray/util/process_helper.cc [hunk 2]
```diff
@@ -79,10 +79,10 @@ void ProcessHelper::RayStart(CoreWorkerOptions::TaskExecutionCallback callback)
     }
   }
 
-  std::unique_ptr<ray::gcs::GlobalStateAccessor> global_state_accessor = nullptr;
+  std::unique_ptr<ray::gcs::GlobalStateAccessor> global_state_accessor =
+      CreateGlobalStateAccessor(redis_address, ConfigInternal::Instance().redis_password);
+  ;
```
Comment (kfstorm) [inline cpp/src/ray/util/process_helper.cc:84]: Duplicated semicolon.

Comment (WangTaoTheTonic) [reply cpp/src/ray/util/process_helper.cc]: ok

File: cpp/src/ray/util/process_helper.cc (hunk 1 above)
Comment (WangTaoTheTonic) [reply cpp/src/ray/util/process_helper.cc]: yes we should. The connect happened in the funciton `CreateGlobalStateAccessor`, so this is a duplicated one. Let's remove it.

**Comment by WangTaoTheTonic** [comment]
Comments addressed. Let's merge it. @kfstorm @wuisawesome

**Review by kfstorm — APPROVED** [review]

**Pull request merged**